tests/integration/test_hostgroup.py

```diff
@@ -0,0 +1,142 @@
+    bundle_path = pack_bundle(from_dir=BUNDLES / "complex_provider", to=tmp_path)
+    return await adcm_client.bundles.create(source=bundle_path)
+
+
+@pytest_asyncio.fixture()
+async def hostprovider(adcm_client: ADCMClient, hostprovider_bundle: Bundle) -> HostProvider:
+    return await adcm_client.hostproviders.create(
+        bundle=hostprovider_bundle, name="Hostprovider name", description="Hostprovider description"
+    )
+
+
+async def test_host_groups(adcm_client: ADCMClient, cluster: Cluster, hostprovider: HostProvider) -> None:
+    for i in range(80):
+        await adcm_client.hosts.create(name=f"test-host-{i}", hostprovider=hostprovider, cluster=cluster)
+
+    for i in range(55):
+        await cluster.action_host_groups.create(name=f"host-group-{i}", description=f"host group description {i}")
+        await hostprovider.config_host_groups.create(name=f"host-group-{i}", description=f"host group description {i}")
+        await cluster.config_host_groups.create(name=f"host-group-{i}", description=f"host group description {i}")
+
+    action_host_group = await cluster.action_host_groups.get(name__eq="host-group-35")
+    config_host_groups_provider = await hostprovider.config_host_groups.get(name__eq="host-group-35")
+    config_host_groups_cluster = await cluster.config_host_groups.get(name__eq="host-group-35")
+
+    await action_host_group.hosts.add(host=await cluster.hosts.filter(name__contains="test-host"))
+    await config_host_groups_provider.hosts.add(host=await adcm_client.hosts.filter(name__contains="test-host"))
+    await config_host_groups_cluster.hosts.add(host=await adcm_client.hosts.filter(name__contains="test-host"))
+
+    await _test_host_group_properties(cluster.action_host_groups)
+    await _test_host_group_accessors(cluster.action_host_groups)
+    await _test_pagination_action_host_group(cluster.action_host_groups)
+
+    await _test_host_group_properties(hostprovider.config_host_groups)
+    await _test_host_group_accessors(hostprovider.config_host_groups)
+    await _test_pagination_action_host_group(hostprovider.config_host_groups)
+
+    await _test_host_group_properties(cluster.config_host_groups)
+    await _test_host_group_accessors(cluster.config_host_groups)
+    await _test_pagination_action_host_group(cluster.config_host_groups)
+
+
+async def _test_host_group_properties(host_group_node: ActionHostGroupNode | ConfigHostGroupNode) -> None:
+    host_group = await host_group_node.get(name__eq="host-group-35")
+    assert host_group.name == "host-group-35"
+    assert host_group.description == "host group description 35"
+
+    hosts_all = await host_group.hosts.all()
+    assert len(hosts_all) == 50
+
+    if isinstance(host_group_node, ConfigHostGroupNode):
+        assert isinstance(host_group, ConfigHostGroup)
+    else:
+        assert isinstance(host_group, ActionHostGroup)
+        assert isinstance(host_group.hosts, HostsInActionHostGroupNode)
```

Comment:
`ConfigHostGroupNode` is expected to have `config` as well as `hosts` node

Reply:
added